### ISSUE TITLE
Can't restore checkpoint across hosts: ERRO[0000] criu failed: type NOTIFY errno 0

### ISSUE THREAD

**Issue by ashish-billore** [checkpoint-restore/criu#1240]
I am able to checkpoint and restore a runc bundle on same VM. 
However, when I copy the checkpoint to another similar VM, restore fails and does not even generate any logs.

What I did is:

1. Start a container:
```
root@criupri:~/criu_test/2host# runc run pri -d &> /dev/null < /dev/null
root@criupri:~# runc list
ID          PID         STATUS      BUNDLE                  CREATED                          OWNER
pri         29035       running     /root/criu_test/2host   2020-10-21T06:23:38.614264868Z   root
root@criupri:~# 


```

Config is: [config_json.txt](https://github.com/checkpoint-restore/criu/files/5413761/config_json.txt)

2. Checkpoint it:
```
root@criupri:~/criu_test/2host# runc checkpoint --manage-cgroups-mode full --leave-running pri
ERRO[0000] container is not destroyed                   
root@criupri:~/criu_test/2host#
```

3. Now I scp the whole runc bundle dir contents (including rootfs, checkpoint dir and config) to 2nd VM, contents are:
```
root@criupri:~/criu_test/2host# ls 
checkpoint  config.json  rootfs 
root@criupri:~/criu_test/2host# 
root@criupri:~/criu_test/2host# scp -r * criubkp:~/criu_test/2host/
```

4. On 2nd VM I attempt restore, it fails and does not even generate any logs, even when it prints there are some logs:

```
root@criubkp:~/criu_test/2host# ls 
checkpoint  config.json  rootfs 
root@criubkp:~/criu_test/2host# 

root@criubkp:~/criu_test/2host# runc restore --manage-cgroups-mode full -d bkp
ERRO[0000] criu failed: type NOTIFY errno 0
log file: /run/runc/bkp/criu.work/restore.log 
criu failed: type NOTIFY errno 0
log file: /run/runc/bkp/criu.work/restore.log

root@criubkp:~/criu_test/2host# ls -ltrh /run/runc/bkp/criu.work/restore.log
ls: cannot access '/run/runc/bkp/criu.work/restore.log': No such file or directory

root@criubkp:~/criu_test/2host# find / -name restore.log
root@criubkp:~/criu_test/2host#
```

What is more interesting is, checkpoint restore works perfectly fine on either of single VMs (restore on the same VM where checkpoint was done).

Can anyone please help get past this issue?

Note: I am trying the same scenario as mentioned [here](https://www.redhat.com/en/blog/container-migration-around-world)

--------------------------------------------------------
System details are:

Setup Details:
```
$criu --version
Version: 3.14

$ runc --version
runc version spec: 1.0.1-dev

$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description: Ubuntu 18.04.4 LTS
Release: 18.04
Codename: bionic

$ uname -a
Linux sctpdev 4.15.0-118-generic #119-Ubuntu SMP Tue Sep 8 12:30:01 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by adrianreber** [comment]
To get the CRIU log files you have to tell runc to put them in another location. Use the option `--work-path` in `runc restore` to put the log files somewhere else. With this it should be possible to get the `restore.log` and understand why restore fails.

**Comment by ashish-billore** [comment]
@adrianreber Thanks for the info, I am able to get the logs now.

Seems criu had some problem with mounting tmp files:
```
(00.033038)      1: Error (criu/mount.c:2012): mnt: Unable to mount tmpfs /tmp/.criu.mntns.7jTNBg/12-0000000000/proc/asound (id=336): No such file or directory
(00.033044)      1: Error (criu/mount.c:2077): mnt: Can't mount at /tmp/.criu.mntns.7jTNBg/12-0000000000/proc/asound: No such file or directory
(00.033047)      1: mnt: Start with 0:/tmp/.criu.mntns.7jTNBg
(00.107293) Error (criu/mount.c:3351): mnt: Can't remove the directory /tmp/.criu.mntns.7jTNBg: Device or resource busy
(00.107311) Error (criu/cr-restore.c:2397): Restoring FAILED.

```
Here is the complete restore failure log
[restore.log](https://github.com/checkpoint-restore/criu/files/5413881/restore.log)

**Comment by adrianreber** [comment]
Not entirely sure, but `/proc/asound` seems to be alsa related. Can you make sure you have the same modules loaded on the destination side. From a quick look at the kernel the module `soundcore` seems to be responsible to create `/proc/asound`. Probably because it is listed as a masked path in `config.json`.

**Comment by ashish-billore** [comment]
@adrianreber thanks for checking this, indeed it was the soundcore module.
At first I enabled the soundcore module ($modprobe soundcore) on the other node, but somehow, restore still did not work.
Once I updated the config.json to remove it, restore worked on other node.

Actually, I used the runc spec tool to generate the config, so just had the default config entries.
I'll try to clean-up the config to remove such unwanted dependencies.

Thanks again for the help!
